### PR TITLE
Bw refactor

### PR DESCRIPTION
**Pull request by muselesscreator**
Update to use redux and re-organize components.

### CODE REVIEW

File: src/editors/data/constants/requests.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { StrictDict } from '../../utils';
+
+export const ReqeustKeys = StrictDict({
```
Comment (connorhaugh) [inline src/editors/data/constants/requests.js:3]: Just to note you can get rid of this here-- unless a `request` is different than a `Reqeust` ;)

File: src/editors/data/redux/thunkActions/app.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  dispatch(requests.saveBlock({
+    content,
+    onSuccess: (response) => {
+      dispatch(actions.app.setSaveResponse(response));
+      returnToUnit();
+    },
+  }));
+};
+
+export default StrictDict({
```
Comment (connorhaugh) [inline src/editors/data/redux/thunkActions/app.js:45]: Why are we exporting these methods in two separate ways (as dict default and as individual constants)? Wouldn't we want to define a usage pattern?

File: src/editors/data/redux/thunkActions/requests.test.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
+import { actions } from 'data/redux';
+import { RequestKeys } from 'data/constants/requests';
+import api from 'data/services/lms/api';
+import * as requests from './requests';
+
+jest.mock('data/services/lms/api', () => ({
```
Comment (connorhaugh) [inline src/editors/data/redux/thunkActions/requests.test.js:6]: Good examples of tests for thunks!

File: src/editors/data/services/lms/api.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import * as urls from './urls';
```
Comment (connorhaugh) [inline src/editors/data/services/lms/api.js:1]: Can we rename this to ...../cms/api.js ? Its a little misleading.

File: src/editors/data/redux/thunkActions/app.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    beforeEach(() => {
+      thunkActions.initialize()(dispatch);
+      [[dispatchedAction]] = dispatch.mock.calls;
+    });
+    it('dispatches initializeApp with locationId and onSuccess', () => {
+      expect(dispatchedAction.initializeApp.locationId).toEqual(locationId);
+      expect(typeof dispatchedAction.initializeApp.onSuccess).toEqual('function');
+    });
+    describe('on success', () => {
+      test('loads oraMetadata, courseMetadata and list data', () => {
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/editors/data/redux/thunkActions/app.test.js:26]: I have not updated any of the tests :-)

File: src/editors/data/redux/thunkActions/app.js (hunk 2 above)
Comment (muselesscreator) [reply src/editors/data/redux/thunkActions/app.js:45]: testing vs consumption

File: src/editors/data/redux/thunkActions/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { StrictDict } from '../../../utils';
+
+import app from './app';
+
+export default StrictDict({
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/editors/data/redux/thunkActions/index.js:5]: yes

File: src/editors/data/services/lms/utils.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
```
Comment (connorhaugh) [inline src/editors/data/services/lms/utils.js:1]: What is the benefit of creating our own module here? We can already mock the AuthenticatedHttpClient, unless I am mistaken?

File: src/editors/data/services/lms/api.js (hunk 4 above)
Comment (muselesscreator) [reply src/editors/data/services/lms/api.js:1]: sure!  This is copying over the service from gradebook.  That is reasonable

File: src/editors/data/redux/requests/selectors.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+export const isFinished = ({ status }) => (
+  [RequestStates.failed, RequestStates.completed].includes(status)
+);
+export const error = (request) => request.error;
+export const errorStatus = (request) => request.error?.response?.status;
+export const errorCode = (request) => request.error?.response?.data;
+
+export const data = (request) => request.data;
+
+export const allowNavigation = ({ requests }) => (
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/editors/data/redux/requests/selectors.js:22]: old holdover.  I removed a bunch of the cruft logic leading to it.  Can be removed at this point.

File: src/editors/data/constants/requests.js (hunk 1 above)
Comment (connorhaugh) [reply src/editors/data/constants/requests.js:3]: Yeah, I was pointing out that you have a misspelled duplicate in this file lol

**Review by connorhaugh — APPROVED** [review]
This isn't complete, but we are merging it and not releasing it so we can add incremental improvements to this large refactor.

**Pull request merged**